src/cinder/Camera.cpp

```diff
@@ -414,6 +414,18 @@ void CameraOrtho::setOrtho( float left, float right, float bottom, float top, fl
 	mProjectionCached = false;
 }
 
+void CameraOrtho::getFarClipCoordinates( vec3 *topLeft, vec3 *topRight, vec3 *bottomLeft, vec3 *bottomRight ) const
+{
+	calcMatrices();
+
+	vec3 viewDirection = normalize( mViewDirection );
+
+	*topLeft		= mEyePoint + (mFarClip * viewDirection) + (mFrustumTop * mV) + (mFrustumLeft * mU);
+	*topRight		= mEyePoint + (mFarClip * viewDirection) + (mFrustumTop * mV) + (mFrustumRight * mU);
+	*bottomLeft		= mEyePoint + (mFarClip * viewDirection) + (mFrustumBottom * mV) + (mFrustumLeft * mU);
+	*bottomRight	= mEyePoint + (mFarClip * viewDirection) + (mFrustumBottom * mV) + (mFrustumRight * mU);
+}
+
```

Comment:
Nothing wrong with this code. But while we're at it: there is a lot of duplicated code in these classes. Perhaps we could write a more generic function to be used internally:
```
protected:
    void getClipCoordinates( float clipDist, float ratio, vec3 *topLeft, vec3 *topRight, vec3 *bottomLeft, vec3 *bottomRight )
    {
        calcMatrices();

        vec3 viewDirection = normalize( mViewDirection );

	*topLeft = mEyePoint + (clipDist * viewDirection) + ratio * ( mFrustumTop * mV + mFrustumLeft * mU );
	*topRight = mEyePoint + (clipDist * viewDirection) + ratio * ( mFrustumTop * mV + mFrustumRight * mU );
	*bottomLeft = mEyePoint + (clipDist * viewDirection) + ratio * ( mFrustumBottom * mV +mFrustumLeft * mU );
	*bottomRight = mEyePoint + (clipDist * viewDirection) + ratio * ( mFrustumBottom * mV + mFrustumRight * mU );
    }
```
and then write our functions like this:
```
virtual void Camera::getNearClipCoordinates( vec3 *topLeft, vec3 *topRight, vec3 *bottomLeft, vec3 *bottomRight ) const {
    getClipCoordinates( mNearClip, mFarClip / mNearClip, topLeft, topRight, bottomLeft, bottomRight );
}
virtual void Camera::getFarClipCoordinates( vec3 *topLeft, vec3 *topRight, vec3 *bottomLeft, vec3 *bottomRight ) const {
    getClipCoordinates( mFarClip, mFarClip / mNearClip, topLeft, topRight, bottomLeft, bottomRight );
}

virtual void CameraOrtho::getNearClipCoordinates( vec3 *topLeft, vec3 *topRight, vec3 *bottomLeft, vec3 *bottomRight ) const override { 
    getClipCoordinates( mNearClip, 1.0f, topLeft, topRight, bottomLeft, bottomRight );
}
virtual void CameraOrtho::getFarClipCoordinates( vec3 *topLeft, vec3 *topRight, vec3 *bottomLeft, vec3 *bottomRight ) const override {
    getClipCoordinates( mFarClip, 1.0f, topLeft, topRight, bottomLeft, bottomRight );
}
```